web/src/enterprise/campaigns/detail/CampaignDetails.tsx

```
@@ -224,8 +224,13 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({
             </div>
         )
     }
+    // error while fetching either campaign or campaignPlan
+    if (!!alertError && (campaign === null || campaignPlan === null)) {
```

`alertError` is documented as for errors from saving or deleting that should be displayed in an alert, while the rest of the campaign is still shown (that's why it is a separate variable). If the error is an error that causes the entire campaign to not be shown, like in this case, the `campaign`/`campaignPlan` would better be a union type `| Error` instead. That enforces it to be handled, essentially preventing this very bug from happening at compile time. See https://docs.sourcegraph.com/dev/typescript/patterns#making-invalid-states-impossible-through-union-types

If the only handling is showing a hero page though, the much easier solution is to just let the error bubble up to the ErrorBoundary, which displays errors on a HeroPage. `useObservable()` does this automatically, so all that is needed is to remove the `catchError()` in line 205-208.
`campaign` also already does this manually without `useObservable()` in line 164 (which means the condition `alertError && campaign === null` actually doesn't seem to ever be true).

---

good catch, thanks